swap_meet/vendor.py

```diff
@@ -1,2 +1,71 @@
+from .item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        inventory = [] if inventory is None else inventory
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        else:
+            self.inventory.remove(item)
+            return item 
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item 
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if (my_item not in self.inventory) or (their_item not in other_vendor.inventory):
+            return False
+        self.remove(my_item)
+        self.add(their_item)
+        other_vendor.remove(their_item)
+        other_vendor.add(my_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+
+        temp_item = self.inventory[0]
+        self.inventory[0] = other_vendor.inventory[0]
+        other_vendor.inventory[0] = temp_item
```

Comment:
We should invoke functions we already used to keep our code concise and DRY.

Since you already have a function that will swap items, we don't need to write additional code that repeats the logic.

```suggestion
        self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```